doc/user_guide/kernel_layer.rst

```diff
@@ -90,24 +90,38 @@ that it allows the metadata to be kept with the code and for it to be
 compilable. In addition, currently all APIs will contain information
 about the arguments in an array called ``meta_args``, a specification
 of what the kernel code iterates over in a variable called
-``iterates_over`` and a reference to the kernel code as a type bound
-procedure. If no type-bound procedure is declared, an interface must be
-specified instead as a reference to the kernel code.
+``iterates_over`` and a reference to the kernel code as a type-bound
+procedure.
+::
+   
+    type, extends(kernel_type) :: integrate_one_kernel 
+      ... 
+      type(...) :: meta_args(...) = (/ ... /) 
+      ... 
+      integer :: ITERATES_OVER = ... 
+      ... 
+      contains 
+      ... 
+      procedure ... 
+      ... 
+    end type integrate_one_kernel 
 
+If no type-bound procedure is declared and a named interface with
+module procedures will be included.     
 ::
 
-    type, extends(kernel_type) :: integrate_one_kernel
-      ...
-      type(...) :: meta_args(...) = (/ ... /)
-      ...
-      integer :: ITERATES_OVER = ...
-      ...
-      contains
-      ...
-      procedure ...
-      ...
-    end type integrate_one_kernel
+    type, extends(kernel_type) :: integrate_one_kernel 
+      ... 
+      type(...) :: meta_args(...) = (/ ... /) 
+      ... 
+      integer :: ITERATES_OVER = ... 
+      ... 
+      contains 
+      ... 
+      procedure ... 
```

Comment:
This one shouldn't have `procedure` in it?

Reply:
Oops. Yes. should have noticed that. Thanks.